CRM/Case/ActionMapping.php

```diff
@@ -0,0 +1,138 @@
+      'id' => CRM_Case_ActionMapping::CASE_MAPPING_ID,
+      'entity' => 'civicrm_case',
+      'entity_label' => ts('Case'),
+      'entity_value' => 'case_type',
+      'entity_value_label' => ts('Case Type'),
+      'entity_status' => 'case_status',
+      'entity_status_label' => ts('Case Status'),
+      'entity_date_start' => 'case_start_date',
+      'entity_date_end' => 'case_end_date',
+    )));
+  }
+
+  /**
+   * @inheritdoc
+   */
+  public function getRecipientListing($type) {
+    if ($type == 'case_roles') {
+      $result = civicrm_api3('RelationshipType', 'get', array(
+        'sequential' => 1,
+        'options' => array('limit' => 0, 'sort' => "label_b_a"),
+      ))['values'];
+      $cRoles = array();
+      foreach ($result as $each) {
+        $cRoles[$each['id']] = $each['label_b_a'];
+      }
+      return $cRoles;
+    }
+    return array();
+  }
+
+  /**
+   * @inheritdoc
+   */
+  public function getRecipientTypes() {
+    return array('case_roles' => 'Case Roles');
+  }
+
+  /**
+   * @inheritdoc
+   */
+  public function createQuery($schedule, $phase, $defaultParams) {
+    $selectedValues = (array) \CRM_Utils_Array::explodePadded($schedule->entity_value);
+    $selectedStatuses = (array) \CRM_Utils_Array::explodePadded($schedule->entity_status);
+
+    $query = \CRM_Utils_SQL_Select::from("civicrm_case c")->param($defaultParams);
+    $query->join('r', "INNER JOIN civicrm_relationship r ON c.id = r.case_id");
+    $query->join('cs', "INNER JOIN civicrm_contact ct ON r.contact_id_b = ct.id");
+    $query->join('ce', "INNER JOIN civicrm_email ce ON ce.contact_id = ct.id");
+
+    if (!empty($selectedValues)) {
+      $query->where("c.case_type_id IN (#caseId)")
+        ->param('caseId', $selectedValues);
+    }
+
+    if (!empty($selectedStatuses)) {
+      $query->where("c.status_id IN (#selectedStatuses)")
+        ->param('selectedStatuses', $selectedStatuses);
+    }
+
+    if (!empty($caseRoles)) {
+      $query->where("r.relationship_type_id IN (#caseRoles)")
+        ->param('caseRoles', $caseRoles);
+    }
+
+    if ($schedule->recipient_listing && $schedule->limit_to) {
+      switch ($schedule->recipient) {
+        case 'case_roles':
+          $caseRoles = (array) \CRM_Utils_Array::explodePadded($schedule->recipient_listing);
+          $query->where("r.relationship_type_id IN (#caseRoles)")
+            ->param('caseRoles', $caseRoles);
+          break;
+      }
+    }
+
+    // $schedule->start_action_date is user-supplied data. validate.
+    if (!array_key_exists($schedule->start_action_date, $this->getDateFields())) {
+      throw new CRM_Core_Exception("Invalid date field");
+    }
+
+    $query['casDateField'] = $schedule->start_action_date;
+    $query['casAddlCheckFrom'] = 'civicrm_case c';
+    $query['casContactIdField'] = 'ct.id';
+    $query['casEntityIdField'] = 'r.case_id';
+    $query['casContactTableAlias'] = 'ct';
+    $query->where('r.is_active = 1 AND c.is_deleted = 0');
```

Comment:
An active relationship is more than checking if the relationship is active alone, We will also need to check the start and end date of the relationship to verify that start date is <= today OR NULL and end date >= today OR NULL.

Reply:
you are right. Fixed this now.